staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -131,6 +131,9 @@ type ClientConfigLoadingRules struct {
 	// WarnIfAllMissing indicates whether the configuration files pointed by KUBECONFIG environment variable are present or not.
 	// In case of missing files, it warns the user about the missing files.
 	WarnIfAllMissing bool
+
+	// WarnIfAllMissingCallback is the warning log callback to use in case of missing files.
+	WarnIfAllMissingCallback func(string, ...interface{})
```

Comment:
I'd probably go with something like:
```golang
WarningHandler func(string)
```

Reply:
Without the formatting directives?

Reply:
yeah, anyone willing to format they'll use `fmt.Sprintf`